### DISCUSSION TITLE
CPU vs GPU - Inference Command Line

### DISCUSSION THREAD

**Discussion by markbland82** [SWivid/F5-TTS#465] [Q&A]
I am running the inference cli command line but it is only using my CPU, is there a reason for this? I have a GPU (3080ti,12gb) which I assume would be able to process the inference much more quickly. Am I missing something? Thanks!

**Comment by SWivid** [comment]
check here https://github.com/SWivid/F5-TTS#:~:text=%23%20Create%20a,few%20options%20below%3A

**Comment by sngraam** [comment]
run infer_gradio - make sure `cuda `is in place check cuda 

`import torch
print("CUDA Available:", torch.cuda.is_available())
print("CUDA Version:", torch.version.cuda)
print("Number of GPUs:", torch.cuda.device_count())
print("GPU Name:", torch.cuda.get_device_name(0) if torch.cuda.is_available() else "No GPU")
`
2. may u use ubuntu (recommeded) if not use it :] if u use window may internel issue i dont know 
3. if u use ubuntu the

goto => task maneger => click processer in top =>  right click python (code)=> click alt + s then see magic

**Comment by markbland82** [comment]
Looks like there were some issues with my venv, likely Torchvision/Audio/Torch causing a break. Torchvision 0.18.0+cu118 appears to have fixed it from what I can tell. 

Thanks for taking the time! F5 is outstanding! Looking forward to future updates @SWivid :) Cheers!
